Webpack config not finding nested node_modules

### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

```
@angular/cli: 1.0.2
node: 6.10.1
os: darwin x64
@angular/common: 4.1.1
@angular/compiler: 4.1.1
@angular/core: 4.1.1
@angular/forms: 4.1.1
@angular/http: 4.1.1
@angular/platform-browser: 4.1.1
@angular/platform-browser-dynamic: 4.1.1
@angular/router: 4.1.1
@angular/cli: 1.0.2
@angular/compiler-cli: 4.1.1
```

### Repro steps.

1. Clone the repo: https://github.com/jshcrowthe/ng-cli-nested-node-modules.git
1. Enter the repo dir
1. Run `npm install`
1. Run `ng serve` 
1. Observe error from `ng serve` command


### The log given by the failure.

```log
ERROR in ./~/firebase/app/shared_promise.js
Module not found: Error: Can't resolve 'promise-polyfill' in '$PATH_TO_REPO/nested-node-module/node_modules/firebase/app'
 @ ./~/firebase/app/shared_promise.js 22:35-62
 @ ./~/firebase/app/firebase_app.js
 @ ./~/firebase/app.js
 @ ./src/app/app.component.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```


### Desired functionality.
Webpack should compile this module and it's nested children correctly


### Mention any other details that might be useful.
n/a

--- a/packages/@angular/cli/models/webpack-configs/common.ts
+++ b/packages/@angular/cli/models/webpack-configs/common.ts
@@ -67,7 +67,7 @@ export function getCommonConfig(wco: WebpackConfigOptions) {
     devtool: buildOptions.sourcemaps ? 'source-map' : false,
     resolve: {
       extensions: ['.ts', '.js'],
-      modules: [nodeModules],
+      modules: ['node_modules', nodeModules],
     },
     resolveLoader: {
       modules: [nodeModules]

--- a/tests/e2e/tests/misc/module-resolution.ts
+++ b/tests/e2e/tests/misc/module-resolution.ts
@@ -1,0 +1,13 @@
+import { appendToFile, prependToFile } from '../../utils/fs';
+import { ng, silentNpm } from '../../utils/process';
+
+
+export default async function () {
+  await silentNpm('install', 'firebase@3.7.8');
+
+  await prependToFile('src/app/app.module.ts', 'import * as firebase from \'firebase\';');
+  await appendToFile('src/app/app.module.ts', 'firebase.initializeApp({});');
+
+  await ng('build');
+  await ng('build', '--aot');
+}

